Storage/SQL/BrowserDB.swift

```diff
@@ -47,6 +47,8 @@ let DBCouldNotOpenErrorCode = 200
 // Version 5 - Added the clients and the tabs tables.
 // Version 6 - Visit timestamps are now microseconds.
 // Version 7 - Eliminate most tables.
+private var queue = dispatch_queue_create("BrowserDBQueue", DISPATCH_QUEUE_CONCURRENT)
```

Comment:
Nit: `let`.

I think this should be serial, not concurrent, until we have the time to be more confident here.

By specifying concurrent, we're allowing multiple operations on this DB to be started serially but complete concurrently. I'm concerned that we might have code somewhere that relies on sequential ordering (e.g., inserting an icon then inserting a bookmark with that icon) that don't strongly impose sequencing.

I *think* Sync is safe with this, because operations are always chained, but I haven't audited with the changed assumption in mind. I don't know about the UI code with any certainty.

More importantly, I'm not confident in how concurrent queues interact with transactions: I would imagine that a queue of operations like

```
  write: begin transaction A
  write: begin transaction B
  …
  write: end transaction A
  …
  write: end transaction B
```

will randomly deadlock, depending on how quickly ops proceed, and that in theory we could end up closing transactions before they start, etc. etc.

Another way to do this, if you want to boost parallelism further, is to allow `run` and friends to take an optional queue argument. But let's give this a whirl first: just adding one additional serial queue might be enough.